src/types.ts

```diff
@@ -16,8 +16,22 @@
 // executing the client function.
 export const FUNCTION_STATUS_HEADER_FIELD = 'X-Google-Status';
 
-export enum SignatureType {
```

Comment:
I don't agree with using string literals and arrays instead of enums. Enums provide benefits over literal values such as better typing, autocompletion, and less code here. We don't plan an arbitrary signature type until the FF contract changes.

This is also a breaking change to the API, not an internal refactor.

Could we discuss a bit more about the tradeoff in the PR / an issue?

Reply:
We're fine changing to the union types, but we shouldn't completely remove the enum as it's an exported interface, unless we have strong reason to introduce a breaking change.

For example, we had a customer complaint about a similar small interface change:
https://github.com/GoogleCloudPlatform/functions-framework-nodejs/issues/216

Reply:
I don't think we give customer an opportunity to pass the value to use in any meaningful way (the signature type can only be provided as an env var or cli flag), so I don't think we should be too concerned about changing this. I see your point about why this is breaking change now. I will updated the conventional commit tag.